java-manta-client-unshaded/src/test/java/com/joyent/manta/client/crypto/MultiCryptoProviderDecryptionTest.java

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright (c) 2019, Joyent, Inc. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.joyent.manta.client.crypto;
+
+import com.joyent.manta.client.MantaClient;
+import com.joyent.manta.client.MantaObjectInputStream;
+import com.joyent.manta.client.MantaObjectResponse;
+import com.joyent.manta.http.MantaHttpHeaders;
+import org.apache.commons.collections4.map.CaseInsensitiveMap;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.io.input.BoundedInputStream;
+import org.apache.commons.lang3.Validate;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.conn.EofSensorInputStream;
+import org.bouncycastle.util.encoders.Hex;
+import org.mockito.Mockito;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.Assert;
+import org.testng.annotations.*;
+import org.testng.annotations.Optional;
+
+import static com.joyent.manta.client.crypto.ExternalSecurityProviderLoader.*;
+import static com.joyent.manta.http.MantaHttpHeaders.*;
+import static org.apache.http.HttpHeaders.*;
+import static org.apache.commons.codec.binary.Hex.encodeHexString;
+
+import javax.crypto.SecretKey;
+import java.io.*;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.security.Provider;
+import java.util.*;
+
+@Test(groups={"encryption-provider"})
+public class MultiCryptoProviderDecryptionTest {
+    private static final Logger LOG = LoggerFactory.getLogger(MultiCryptoProviderDecryptionTest.class);
+    private static final byte[] SECRET_KEY = Hex.decode(
+                "8b30335ba65d1d0619c6192edb15318763d9a1be3ff916aaf46f4717232a504a");
+    private static final List<SupportedCipherDetails> CIPHERS_TO_TEST = Collections.unmodifiableList(Arrays.asList(
+                AesGcmCipherDetails.INSTANCE_128_BIT,
+                AesGcmCipherDetails.INSTANCE_192_BIT,
+                AesGcmCipherDetails.INSTANCE_256_BIT,
+                AesCtrCipherDetails.INSTANCE_128_BIT,
+                AesCtrCipherDetails.INSTANCE_192_BIT,
+                AesCtrCipherDetails.INSTANCE_256_BIT,
+                AesCbcCipherDetails.INSTANCE_128_BIT,
+                AesCbcCipherDetails.INSTANCE_192_BIT,
+                AesCbcCipherDetails.INSTANCE_256_BIT));
+    private static byte[] expectedContent;
```

Comment:
What is this field for and why is it static?

Reply:
Removed static scope, ```expectedContent``` is the original data input which is being compared against the decrypted output of the original file after going through encryption/decryption.